lib/subordinateio.c

```diff
@@ -770,6 +799,8 @@ int list_owner_ranges(const char *owner, enum subid_type id_type, struct subid_r
 	enum subid_status status;
 	int count = 0;
 	struct subid_nss_ops *h;
+	char *id = malloc(sizeof(char) * ID_SIZE);
```

Comment:
No check for NULL?

Any reason not to just do 'char id[ID_SIZE]' ?

Reply:
I've changed it to `char id[ID_SIZE]`